stdlib/public/core/OutputStream.swift

```diff
@@ -330,9 +330,19 @@ internal struct _Stdout : OutputStreamType {
   }
 
   mutating func write(string: String) {
-    // FIXME: buffering?
-    // It is important that we use stdio routines in order to correctly
-    // interoperate with stdio buffering.
+    if string.isEmpty { return }
+
+    if string._core.isASCII {
+      defer { _fixLifetime(string) }
+
+      let result = _swift_stdlib_fwrite_stdout(
+        UnsafePointer(string._core.startASCII), string._core.count, 1)
+      if result != 1 {
+        fatalError("fwrite() returned \(result), expected 1")
```

Comment:
on Tue Mar 01 2016, Kevin Ballard <notifications-AT-github.com> wrote:

>> @@ -330,9 +330,19 @@ internal struct _Stdout : OutputStreamType {
>>    }
>>  
>>    mutating func write(string: String) {
>> -    // FIXME: buffering?
>
>> -    // It is important that we use stdio routines in order to correctly
>> -    // interoperate with stdio buffering.
>> +    if string.isEmpty { return }
>> +
>> +    if string._core.isASCII {
>> +      defer { _fixLifetime(string) }
>> +
>> +      let result = _swift_stdlib_fwrite_stdout(
>> +        UnsafePointer(string._core.startASCII), string._core.count, 1)
>> +      if result != 1 {
>> +        fatalError("fwrite() returned \(result), expected 1")
>
> And aborting the program over a mere failure to print to stdout seems
> rather extreme. If we can't print to stdout, presumably that means
> there's nothing listening on the other end of the pipe anymore, so
> it's not like there's anything that would have been able to see our
> output if we did successfully print. 

This sounds like a classic case of “don't make it a precondition if it
can't reasonably/efficiently be checked for.” I agree with Kevin; at
most we should be throwing when print fails, but I think probably we'd
want a separate function for that kind of functionality.  After all,
`print` isn't set up to deal with disk full or other kinds of I/O
errors.  I see it as as a casual logging tool.

-- 
-Dave